ui/app/[locale]/(authorized)/kafka/[kafkaId]/nodes/rebalances/RebalanceTable.tsx

```diff
@@ -0,0 +1,294 @@
+  onFilterStatusChange: (status: RebalanceStatus[] | undefined) => void;
+} & Pick<
+  TableViewProps<RebalanceList, (typeof RebalanceTableColumns)[number]>,
+  "isColumnSortable" | "onPageChange" | "onClearAllFilters"
+>;
+export function RebalanceTable({
+  baseurl,
+  isColumnSortable,
+  rebalanceList,
+  rebalancesCount,
+  onApprove,
+  onStop,
+  filterName,
+  filterStatus,
+  filterMode,
+  onFilterNameChange,
+  onFilterStatusChange,
+  onFilterModeChange,
+  page,
+  perPage,
+  onPageChange,
+  onRefresh,
+}: RebalanceTabelProps) {
+  const t = useTranslations("Rebalancing");
+  return (
+    <TableView
+      itemCount={rebalancesCount}
+      page={page}
+      perPage={perPage}
+      onPageChange={onPageChange}
+      isRowExpandable={() => true}
+      columns={RebalanceTableColumns}
+      isColumnSortable={isColumnSortable}
+      data={rebalanceList}
+      emptyStateNoData={<></>}
+      emptyStateNoResults={<></>}
+      ariaLabel={"Rebalance brokers"}
+      renderHeader={({ Th, column, key }) => {
+        switch (column) {
+          case "name":
+            return (
+              <Th key={key} width={30} dataLabel={"Rebalance"}>
+                {t("rebalance_name")}
+              </Th>
+            );
+          case "status":
+            return (
+              <Th key={key} dataLabel={"Status"}>
+                {t("status")}
+              </Th>
+            );
+          case "createdAt":
+            return (
+              <Th key={key} dataLabel={"Consumer groups"}>
+                {t("created_at")}
+              </Th>
+            );
+        }
+      }}
+      renderCell={({ Td, column, row, key }) => {
+        switch (column) {
+          case "name":
+            return (
+              <Td key={key} dataLabel={"Rebalance Name"} width={10}>
+                <Link href={`${baseurl}/${row.id}`}>
+                  <Badge key={key}>{t("cr_badge")}</Badge> {row.attributes.name}
+                </Link>
+              </Td>
+            );
+          case "status":
+            return (
+              <Td key={key} dataLabel={"Status"}>
+                {StatusLabel[row.attributes.status]}
+              </Td>
+            );
+          case "createdAt":
+            return (
+              <Td key={key} dataLabel={"Created At"}>
+                {row.attributes.creationTimestamp}
+              </Td>
+            );
+        }
+      }}
+      renderActions={({ row, ActionsColumn }) => (
+        <ActionsColumn
+          items={[
+            {
+              title: t("approve"),
+              onClick: () => onApprove(row),
+              isDisabled: row.meta?.allowedActions.includes("approve"),
+            },
+            {
+              title: t("refresh"),
+              onClick: () => onRefresh(row),
+            },
+            {
+              title: t("stop"),
+              onClick: () => onStop(row),
+              isDisabled: !row.meta?.allowedActions.includes("approve"),
```

Comment:
```suggestion
              isDisabled: !row.meta?.allowedActions.includes("stop"),
```

Reply:
I have implemented the logic according to the design: if 'approve' is present in the allowedActions list, the 'Stop' button will be enabled. If 'approve' is absent in the allowed actions, the 'Stop' button will be disabled, and the 'Approve' button will be enabled.

Reply:
<img width="409" alt="Screenshot 2024-10-03 at 4 37 47 PM" src="https://github.com/user-attachments/assets/a184eb00-4d53-42f3-bd23-802afd3d56b9">

Reply:
> if 'approve' is present in the allowedActions list, the 'Stop' button will be enabled. If 'approve' is absent in the allowed actions, the 'Stop' button will be disabled, and the 'Approve' button will be enabled.

I don't think these make sense. The Stop option should only be enabled when `stop` is one of the `allowedActions`. Similarly, the Approve option should be enabled when `approve` is one of the `allowedActions`. Also, the same for the Refresh option. These are driven by the current state, as defined by Strimzi's [`KafkaRebalanceState`](https://github.com/strimzi/strimzi-kafka-operator/blob/637d93483ba1ff1a9277e437f04b40d31e09b0ba/api/src/main/java/io/strimzi/api/kafka/model/rebalance/KafkaRebalanceState.java#L9)

Reply:
updated